azkaban-common/src/main/java/azkaban/executor/container/ContainerizedExecutionManager.java

```diff
@@ -0,0 +1,383 @@
+   * container.
+   */
+  private class QueueProcessorThread extends Thread {
+
+    private final long queueProcessorWaitInMS;
+    private final Props azkProps;
+    private volatile boolean shutdown = false;
+    private volatile boolean isActive = true;
+    private ExecutorService executorService;
+    private ExecutorLoader executorLoader;
+    private boolean executionsBatchProcessingEnabled;
+    private int executionsBatchSize;
+
+    public QueueProcessorThread(final Props azkProps, final ExecutorLoader executorLoader) {
+      this.azkProps = azkProps;
+      this.executorLoader = executorLoader;
+      setActive(
+          this.azkProps.getBoolean(Constants.ConfigurationKeys.QUEUEPROCESSING_ENABLED, true));
+      this.queueProcessorWaitInMS =
+          this.azkProps.getLong(Constants.ConfigurationKeys.QUEUE_PROCESSOR_WAIT_IN_MS, 1000);
+      this.executionsBatchProcessingEnabled = azkProps
+          .getBoolean(ContainerizedExecutionManagerProperties.CONTAINERIZED_EXECUTION_BATCH_ENABLED,
+              false);
+      this.executionsBatchSize =
+          azkProps
+              .getInt(ContainerizedExecutionManagerProperties.CONTAINERIZED_EXECUTION_BATCH_SIZE,
+                  10);
+
+      this.executorService = Executors.newFixedThreadPool(azkProps.getInt(
+          ContainerizedExecutionManagerProperties.CONTAINERIZED_EXECUTION_PROCESSING_THREAD_SIZE,
+          10));
+      this.setName("Containerized-QueueProcessor-Thread");
+    }
+
+    @Override
+    public void run() {
+      // Loops till QueueProcessorThread is shutdown
+      while (!this.shutdown) {
+        synchronized (this) {
+          try {
+            // start processing queue if active, otherwise wait for sometime
+            if (this.isActive) {
+              processQueuedFlows();
+            }
+            wait(queueProcessorWaitInMS);
+          } catch (final Exception e) {
+            ContainerizedExecutionManager.logger.error(
+                "QueueProcessorThread Interrupted. Probably to shut down.", e);
+          }
+        }
+      }
+    }
+
+    /* Method responsible for processing the non-dispatched flows */
+    private void processQueuedFlows() throws ExecutorManagerException {
+      Set<Integer> executionIds =
+          executorLoader.selectAndUpdateExecutionWithLocking(this.executionsBatchProcessingEnabled, this.executionsBatchSize,
+              Status.DISPATCHING);
+
+      for (int executionId : executionIds) {
+        Runnable worker = new ExecutionDispatcher(executionId);
+        executorService.execute(worker);
+      }
+    }
+
+    public boolean isActive() {
+      return this.isActive;
+    }
+
+    public void setActive(final boolean isActive) {
+      this.isActive = isActive;
+      ContainerizedExecutionManager.logger
+          .info("QueueProcessorThread active turned " + this.isActive);
+    }
+
+    public void shutdown() {
```

Comment:
Will be good to put a logging message in the shutdown path

Reply:
Added.